libs/vm/CMakeLists.txt

```diff
@@ -20,3 +20,7 @@ target_link_libraries(fetch-vm
                              fetch-math
                              fetch-core
                              fetch-ledger)
+
+if (_is_clang_compiler)
+  target_compile_options(fetch-vm PRIVATE -Wshadow)
```

Comment:
It is not clear if this has been enabled for all libraries. If it has is it worth simple adding the configuration at the top level?

Reply:
That was my intention for the next PR - not all libraries are enabled which I would do next. Then think about GCC enabling (but probably not)